src/libre/dialect/Makefile

```diff
@@ -30,6 +30,11 @@ DFLAGS.${src} += -D LX_HEADER='"lexer.h"'
 .for src in ${SRC:Msrc/libre/dialect/${dialect}/parser.c}
 CFLAGS.${src} += -D DIALECT='${dialect}'
 DFLAGS.${src} += -D DIALECT='${dialect}'
+
+.if ${dialect:T:Mpcre}
+CFLAGS.${src} += -D PCRE_DIALECT=1
+DFLAGS.${src} += -D PCRE_DIALECT=1
+.endif
 .endfor
```

Comment:
Not a huge fan of having centralised code know about dialect-specific things. Normally I would try to keep that under src/libre/dialect/pcre/re_dialect_pcre.c (as I did for dialect-specific handling for character classes), but that wouldn't work for the parser internals here. So I see why you're doing it, and I don't have any other approaches which make sense at the moment.

Reply:
Yeah, I started out putting a function in `re_dialect_pcre.c`, but ... it needs to know about the internals of `struct act_state` and `struct lex_state`, and it seemed like a huge mess.

This is at least a small, contained mess.